### ISSUE TITLE
[BOOM-2065] Decimal support removed from currency

### ISSUE THREAD

**Issue by prestonBot** [PrestaShop/PrestaShop#10017]
> This issue has been migrated from the Forge. Read the [original ticket here](http://forge.prestashop.com/browse/BOOM-2065).

- _**Reporter:**_ bellini13
- _**Created at:**_ Wed, 21 Dec 2016 15:34:27 +0100

<p>Not all currencies in this world support decimals, and with the removal of decimals from the currency, how does a store support multiple currencies (ones with and without decimals).</p>

<p>Take ISK currency for an example, this currency does not use decimals...</p>

<p>This is a huge oversight and needs to be addressed</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-138309).

- _**Author:**_ vincentbz
- _**Created at:**_ Mon, 2 Jan 2017 19:32:28 +0100

<p>Ticket en cours d'étude.</p>

<p>Définir si c'est le CLDR qui prend en charge le nombre de décimales. A priori non, mais c'est à confirmer.<br/>
Dans ce cas, nous devons, dans chaque pack de loc, définir quel est le nombre de décimales pour la monnaie. C'est une norme ISO, facile à trouver.<br/>
Enfin, on doit réparer le truc, ce qui est un peu chiant, car ça nécessite également de réparer les prix et les arrondis.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-139518).

- _**Author:**_ bellini13
- _**Created at:**_ Tue, 24 Jan 2017 16:15:23 +0100

<p>When is this issue (and other currency related issues) going to be addressed?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-140608).

- _**Author:**_ vincentbz
- _**Created at:**_ Tue, 7 Feb 2017 11:28:41 +0100

<p>Hi bellini13 bellini13,</p>

<p>It may not be fixed in the 1.7.1 version but more probably in the following one.<br/>
Thanks for your patience <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>Regards</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-140609).

- _**Author:**_ vincentbz
- _**Created at:**_ Tue, 7 Feb 2017 11:32:12 +0100

<p><b>Fix</b><br/>
In the BO, there is an option in "Configure / Shop parameters / General" called "Number of decimals". It allows to change the number of decimals applied on the Front Office.<br/>
This option doesn't work anymore.</p>

<p>It should have the same behaviour like in 1.6, be careful to the round mode.</p>

<p>Thanks</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-140638).

- _**Author:**_ bellini13
- _**Created at:**_ Tue, 7 Feb 2017 14:33:25 +0100

<p>That is not an acceptable 'workaround'.  That BO option would apply to every single currency in the Shop.  A merchant can have multiple currencies in their shop (for example EUR, USD, ISK).  Some of them require decimals and some of them do not.  </p>

<p>Decimal support feature needs to be re-included in Prestashop, as this behavior existing going back to at least PS v1.4.</p>

<p>I don't understand why Prestashop has to remove features that have existed and worked for so many years.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-140666).

- _**Author:**_ vincentbz
- _**Created at:**_ Tue, 7 Feb 2017 16:03:45 +0100

<p>Thanks for your feedback. I think this fix will be a first step. As you say, it's not the perfect behaviour as it should be specific for each currency. </p>

<p>In 1.6 we can globally chose the number of decimals and for each currency if we want to display decimals. In 1.7 we've started to include the CLDR and some options are related to it, some others not. This is why some features disappeared (maybe too many). But we still need to massively improve and fix it.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-144875).

- _**Author:**_ alexeven
- _**Created at:**_ Wed, 12 Apr 2017 19:05:18 +0200

<p><a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=vincentbz" class="user-hover" rel="vincentbz">Vincent Beudez</a> J'enlève le ticket de la liste To-Do 1.7, ça sera pris en compte dans la réimplémentation du CLDR (I'm on it), là tout de suite on ne va rien faire là dessus.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-144883).

- _**Author:**_ bellini13
- _**Created at:**_ Wed, 12 Apr 2017 21:20:09 +0200

<p>What information could possibly be needed?  Why are we using software (CLDR) that is not correct?  And why are we continuing to remove features from PS that have been supported for years.  </p>

<p>We need to find a way forward, not backwards...</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-144909).

- _**Author:**_ alexeven
- _**Created at:**_ Thu, 13 Apr 2017 11:51:43 +0200

<p>Hi bellini13 bellini13,</p>

<p>This is something we are working on, and I changed the status to "Need more info" as I need to provide more info to the team before we can fully fix this. It's more like a reminder for myself ^^</p>

<p>If we need more details from you, I'll let you know <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>Thank you!</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-149890).

- _**Author:**_ musicmaster
- _**Created at:**_ Sun, 9 Jul 2017 21:41:43 +0200

<p>Still not fixed in 1.7.2. What a shame!</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-150735).

- _**Author:**_ alexeven
- _**Created at:**_ Mon, 24 Jul 2017 18:19:41 +0200

<p>Hello <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=musicmaster" class="user-hover" rel="musicmaster">Wim Roffel</a>, <br/>
This is planned for version 1.7.3.<br/>
Thanks for your understanding!</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-154539).

- _**Author:**_ andrejstas
- _**Created at:**_ Thu, 28 Sep 2017 08:35:12 +0200

<p>When can we expect 1.7.3 with this problem fixed?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-155207).

- _**Author:**_ stgoddv
- _**Created at:**_ Thu, 12 Oct 2017 14:08:14 +0200

<p>Hi. <br/>
Can someone tell us if this it's going to be resolved by 1.7.3?<br/>
I thinks it's a big problem. Here in Chile, for example, we don't use decimals.<br/>
The products simply doesn't look good at the front page, and cause confusion to the customers.<br/>
Please correct that, we'll be very grateful.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-157873).

- _**Author:**_ decormoon
- _**Created at:**_ Tue, 5 Dec 2017 11:19:31 +0100

<p>Dear Prestashop,</p>

<p> </p>

<p>I also faced this problem with version 1.7.2.4 in my country(Vietnamese)</p>

<p>Could you please confirm the fix will be included in next release 1.7.3 ?</p>

<p> </p>

<p>Thanks</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-157875).

- _**Author:**_ webfantasy.it
- _**Created at:**_ Tue, 5 Dec 2017 11:28:32 +0100

<p>I the versione 1.7.3,</p>

<p>will you reintroduce also the currency format that was always present up to 1.6 and that you removed from 1.7?</p>

<p>Thank you!</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-158056).

- _**Author:**_ decormoon
- _**Created at:**_ Fri, 8 Dec 2017 06:44:38 +0100

<p>Hi all,</p>

<p>This problem seems to be duplicated with : <a href="http://forge.prestashop.com/browse/BOOM-2983" class="external-link" rel="nofollow">http://forge.prestashop.com/browse/BOOM-2983</a> </p>

<p>It's great if we can have fix in next SW release 1.7.3</p>

<p> </p>

<p>Thanks.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-158057).

- _**Author:**_ decormoon
- _**Created at:**_ Fri, 8 Dec 2017 06:45:33 +0100

<p>This is really serious problem because it displays to user via front-office.</p>

<p>Hope it can be fixed in 1.7.3</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-159494).

- _**Author:**_ lilianamc
- _**Created at:**_ Mon, 25 Dec 2017 17:21:00 +0100

<p>It wan't fix in PrestaShop v1.7.3.0 Beta 1 <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-159635).

- _**Author:**_ louise bonnard
- _**Created at:**_ Thu, 28 Dec 2017 11:42:23 +0100

<p>Hi everyone, as I said in <a href="http://forge.prestashop.com/browse/BOOM-3460" class="external-link" rel="nofollow">BOOM-3460</a>, the CDLR still requires a lot of improvement for now, and this is why it did not integrate the 1.7.3. We have to continue working on it to have it all perfect. Let's hope it will be ready for version 1.7.4. We are sorry for the inconvenience but we prefer to get the CLDR working perfectly fine for the next version rather than releasing it with many bugs for the 1.7.3.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-159644).

- _**Author:**_ bellini13
- _**Created at:**_ Thu, 28 Dec 2017 14:16:55 +0100

<p>1 year later and still not fixed...   here is a crazy thought, why did Prestashop start using a component (CLDR) if it is so buggy.  And if it continues to be buggy, why continue to use it???  Why not just revert Prestashop back to the previous functionality (which was working perfectly fine) so that PS v1.7 can work for businesses, and once CLDR is in a working state, then feel free to reintroduce it.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-159650).

- _**Author:**_ lilianamc
- _**Created at:**_ Thu, 28 Dec 2017 14:32:58 +0100

<p>Guys, </p>

<p>I found how to fix it (I'm so happy), just follow the steps >> <a href="https://www.prestashop.com/forums/topic/608078-guide-how-to-configurecustomize-your-currency-in-prestashop-17/" class="external-link" rel="nofollow">https://www.prestashop.com/forums/topic/608078-guide-how-to-configurecustomize-your-currency-in-prestashop-17/</a></p>

<p> </p>

<p>I love Prestashop 1.7 </p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-167813).

- _**Author:**_ dennispw
- _**Created at:**_ Wed, 16 May 2018 10:12:32 +0200

<p>Still present in 1.7.3.2</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-167817).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Wed, 16 May 2018 10:16:18 +0200

<p>Hi Dennis,</p>

<p>This issue is not fixed yet.<br/>
There are some major issues to solve before this one.<br/>
But PrestaShop is an open source project, so it can be solved before if someone submits a pull request to solve it.<br/>
Thanks for your understanding!</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-167838).

- _**Author:**_ dennispw
- _**Created at:**_ Wed, 16 May 2018 11:34:59 +0200

<p>I also tried it in 1.7.4.x, just to check. Not yet fixed as you said. Never the less it would be really great if this particular setting could be added back to the currency label. Some currencies might need decimals where others might not. A global setting could work in some cases but in most cases it will not.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-168356).

- _**Author:**_ dennispw
- _**Created at:**_ Thu, 24 May 2018 14:18:50 +0200

<p>Is there a specific version you're aming at in which you have this fixed?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-168357).

- _**Author:**_ khouloud.belguith
- _**Created at:**_ Thu, 24 May 2018 14:33:07 +0200

<p>Hi Dennis,</p>

<p>No, this issue is not fixed yet in any version PS1.7.</p>

<p>Best regards, Khouloud</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-168468).

- _**Author:**_ marion_francois
- _**Created at:**_ Fri, 25 May 2018 15:20:14 +0200

<p><a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=pablo.borowicz" class="user-hover" rel="pablo.borowicz">Pablo Borowicz</a></p>

<p>Will it be solved with CLDR ?</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-169429).

- _**Author:**_ kapselj
- _**Created at:**_ Sun, 10 Jun 2018 11:10:08 +0200

<p>Hi,</p>

<p>Has anybody discovered any temporary fix for that?</p>

<p>We are unable to use zero decimal currencies with PS? It refers to popular currencies as JPY, KRW and more  <img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>best regards,</p>

<p>Karol</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-169431).

- _**Author:**_ lilianamc
- _**Created at:**_ Sun, 10 Jun 2018 14:29:47 +0200

<p>I'm from Colombia and we can't use PS 1.7. </p>

<p>We don't use decimals, and customers will not buy if they see a website with decimals. </p>

<p>On the top of that our colombian payment gateways don't support decimals. </p>

<p><img class="emoticon" src="http://forge.prestashop.com/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/> </p>

<p> </p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-169435).

- _**Author:**_ dennispw
- _**Created at:**_ Mon, 11 Jun 2018 09:08:02 +0200

<p>Don't think so. If it's critical you can still get around it by manually modifying the language/currency files as explained in this forumpost: <a href="https://www.prestashop.com/forums/topic/574893-how-to-disable-decimal-number-prestashop-1702/#comment-2503460" class="external-link" rel="nofollow">https://www.prestashop.com/forums/topic/574893-how-to-disable-decimal-number-prestashop-1702/#comment-2503460</a></p>

<p>Remember that this is not an official fix/tip so be sure to make backups prior to any modifications.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-170376).

- _**Author:**_ krossdal
- _**Created at:**_ Wed, 27 Jun 2018 01:03:15 +0200

<p>Are there any news on this issue? <br/>
19th of September 2017 <a href="http://forge.prestashop.com/secure/ViewProfile.jspa?name=prasanth" class="user-hover" rel="prasanth">Prasanth Selvarasa</a> said that you where working on it in <a href="http://forge.prestashop.com/browse/BOOM-3853" title="Settings for product price decimal places not working" class="issue-link" data-issue-key="BOOM-3853"><del>BOOM-3853</del></a>.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-170385).

- _**Author:**_ marion_francois
- _**Created at:**_ Wed, 27 Jun 2018 10:59:02 +0200

<p>Hi everyone,</p>

<p>It should be fixed in 1.7.5.0 with CLDR</p>

<p>Regards</p>

<p> </p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-170391).

- _**Author:**_ pablo.borowicz
- _**Created at:**_ Wed, 27 Jun 2018 11:10:06 +0200

<p>> 1 year later and still not fixed...   here is a crazy thought, why did Prestashop start using a component (CLDR) if it is so buggy.</p>

<p>CLDR is a Unicode standard, not a component. </p>

<p>Why CLDR? Because formatting numbers in several localizations is incredibly complex. More than you can imagine. CLDR specifies how numbers are formatted (and other stuff like dates and such but we aren't using that right now) in all languages and cultures. PrestaShop 1.6 did this right, but only in some cultures.</p>

<p>The current implementation of CLDR in the 1.7 Core uses a library whose support was dropped by the maintainer after it was included in the core. Another problem of this implementation is that you can't create new currencies that are not standardized by CLDR.</p>

<p>Since last year we have been working on a reimplementation of CLDR that not only would use the latest standard, but it will also allow us to create custom currencies (like crypto currencies). This proved to be a lot more challenging than we initially thought. We were almost ready to include the first part of this rework in 1.7.4 but we found some blocking issues elsewhere that prevented us from including it for that version.</p>

<p>> And if it continues to be buggy, why continue to use it???  Why not just revert Prestashop back to the previous functionality (which was working perfectly fine) so that PS v1.7 can work for businesses, and once CLDR is in a working state, then feel free to reintroduce it.</p>

<p>We can't revert to the old implementation because it was "buggy" in a different way. The solution is to finish the job with the new implementation.</p>

**Comment by prestonBot** [comment]
> This comment has been migrated from the Forge. Read the [original comment here](http://forge.prestashop.com/browse/BOOM-2065#comment-174357).

- _**Author:**_ bellini13
- _**Created at:**_ Fri, 17 Aug 2018 12:58:18 +0200

<p>truly unbelievable that this is delayed yet again...</p>

<p>CLDR is a component that is embedded in Prestashop, and defective.  But instead of acknowledging that and rolling back to the way things were (which worked perfectly fine since PS v1.2), they would rather deprive the user base of the functionality. </p>

**Comment by peranderssjostrom** [comment]
I have also a very annoying problem in the same topic. The PS 1.6.1.20 didn't have it.
https://github.com/PrestaShop/PrestaShop/issues/10238

**Comment by dennispw** [comment]
Any update on this?

**Comment by colinegin** [comment]
@tomlev and @MatShir , do we agree that this issue will be fixed with CLDR (rework part 2)?

**Comment by joelprestaworks** [comment]
Do we have any status update for this? It's a URGENT matter!

**Comment by MatShir** [comment]
We are sorry for the delay but editing/customizing currency features (included decimal management) is on the roadmap 1.7.6. 

The first part of the feature #[10052](https://github.com/PrestaShop/PrestaShop/issues/10052) is almost done which the implementation of the CLDR (the data needed to manage a maximum of currencies),  is now waiting reviewing and go through the QA. Check the PR #10455 try it to see if it partially fixes your issues on your currency display.

For the customizing part, now we are reworking (again I know) on the specification to be sure we don't miss any case and on the design. Find the issue here #11660 working in progress. But it is also part of the 1.7.6 roadmap.

**Comment by marionf** [comment]
I close it as CLDR has been merged for 1.7.6

**Comment by Hlavtox** [comment]
@marionf What effect does the merging have on currencies without decimal places?

1.7.6 beta 1 and I'm still unable to disable showing of decimal places. The value in General settings doesn't work, on currency edit page, there is no option. This whole thing is just unusable from since 1.7.

**Comment by marionf** [comment]
Hi @Hlavtox 

It's planned to remove the number of decimal parameter in Shop parameters > General, as you can see here: https://github.com/PrestaShop/PrestaShop/issues/12223
And it's also planned to add this feature in CLDR part 2: https://github.com/PrestaShop/PrestaShop/issues/11660

This is why this issue is closed, because there are 2 others with more details & specifications

**Comment by khouloudbelguith** [comment]
Duplicate of #12223
